### ISSUE TITLE
Require pkg isn't ordering states properly in 2015.5.3

### ISSUE THREAD

**Issue by ryanwalder** [saltstack/salt#26241]
After updating to 2015.5.3 a lot of our requisites have stopped working as expected which breaks salt in a spectacular way when using git because git doesn't exist yet:

```
deployment-key: 
  file.managed:
    - name: /etc/ssh/deployment-key
    - contents_pillar: deployment-key
    - user: root
    - group: root
    - mode: 600

get-repo:
  git.latest:
    - name: git@git.blah.com:reponame
    - target: /opt/repo
    - identity: /etc/ssh/deployment-key
    - force_checkout: True
    - user: root
    - require:
      - file: deployment-key
      - pkg: install-git

install-git:
  pkg.latest:
      - name: git-core
```

This doesn't re-order the states to make `install-git` come before `get-repo` as you would expect.

```
vagrant@ilikecheese:~$ salt-minion --versions
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: Not Installed
               Tornado: 4.2.1
 Debian source package: 2015.5.3+ds-1trusty1
```

**Comment by ryanwalder** [comment]
OK, retested with a git version of 2015.8 and it's fixed!

```
vagrant@ilikecheese:~$ salt-minion --versions
Salt Version:
           Salt: 2015.8.0rc3-36-g1492a14
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
       pycrypto: 2.6.1
        timelib: 0.2.4
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-57-generic
         system: Ubuntu 14.04 trusty
```

**Comment by bbinet** [comment]
Is it also actually fixed in the 2015.5 branch?

**Comment by ryanwalder** [comment]
I've not tested with the git 2015.5.3 to be honest, I would guess the packages are made whenever that version is cut though and 2015.5.3 looks to be the last of the 2015.5 branch.

I'll see if I can test it when I'm in the office.

**Comment by bbinet** [comment]
This was just to make sure this issue will not be present in the next release (2015.5.4) of the 2015.5 branch.

**Comment by ryanwalder** [comment]
Ah, didn't realise there would be another point release for 2015.5, will test it today and re-open if needed.

**Comment by bbinet** [comment]
Thanks.

**Comment by ryanwalder** [comment]
Looks like it's fixed in the 2015.5 branch, safe to leave this closed.

**Comment by bbinet** [comment]
Great, thanks for testing!
